devices/Hx711/Hx711.cs

```diff
@@ -0,0 +1,225 @@
+            // gain factor from property
+            readSamplePulseTrain = new byte[]
+            {
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                (byte)Gain
+            };
+        }
+
+        /// <summary>
+        /// Read weight from the scale.
+        /// </summary>
+        /// <returns></returns>
+        public double Read()
+        {
+            return ReadValue() - Offset;
+        }
+
+        /// <summary>
+        /// Sets the <see cref="Offset"/> value for tare weight.
+        /// </summary>
+        public void Tare()
+        {
+            Offset = ReadValue();
+        }
+
+        /// <summary>
+        /// Puts the device into power down mode.
+        /// </summary>
+        public void PowerDown()
+        {
+            // transition of CLK signal low > high with 60us 
+            _spiDevice.Write(new ushort[] { 0xFFFF, 0xFFFF, 0xFFFF });
+        }
+
+        /// <summary>
+        /// Wakes up the device from power down mode.
+        /// </summary>
+        public void PowerUp()
+        {
+            // only required if the device is in power down mode
+            var currentDout = _spiDevice.ReadByte();
+
+            if (currentDout != 0)
+            {
+                // transition of CLK signal high > low to wake-up device
+
+                // set gain factor
+                SetChannelAndGainFactor();
+            }
+        }
+
+        private int ReadValue()
+        {
+            Debug.WriteLine("INFO: Reading sample.");
+
+            // setup buffer to drive PD_SCK
+            SpanByte clkTrain = new(readSamplePulseTrain);
+
+            // setup buffer to hold data read from DOUT
+            SpanByte readBuffer = new(new byte[7]);
+
+            // setup array to hold readings for averaging
+            int[] values = new int[SampleAveraging];
+
+            for (int round = 0; round < SampleAveraging; round++)
+            {
+                // need to way for the next conversion
+                if (WaitForConversion())
+                {
+                    // perform SPI transaction
+                    _spiDevice.TransferFullDuplex(clkTrain, readBuffer);
+
+                    values[round] = ParseRawData(readBuffer);
+                }
+            }
+
+            return ComputeAverage(values);
+        }
+
+        private bool WaitForConversion()
+        {
+            Debug.WriteLine("INFO: Setup sampling to detect that a sample is ready");
+
+            var currentDout = _spiDevice.ReadByte();
+
+            while (currentDout != 0)
+            {
+                Thread.Sleep(10);
+                currentDout = _spiDevice.ReadByte();
+            }
+
+            return true;
+        }
+
+        private void SetChannelAndGainFactor()
+        {
+            // send N clock pulses according to the set gain factor
```

Comment:
I'm not sure I understand the comment regarding the code. It just seems that you write a byte, so I may have miss something from the comment